std/range/primitives.d

```diff
@@ -292,9 +292,9 @@ void put(R, E)(ref R r, E e)
     else static if (is(typeof(doPut(r, [e]))) && !isDynamicArray!R)
     {
         if (__ctfe)
-            doPut(r, [e]);
+            doPut(r, [e]); // TODO: breaks @nogc inference
```

Comment:
Possible way to avoid array literal allocation in CTFE:

```d
if (__ctfe)
{
    auto[$] arr = [e];
    doPut(r, arr[]);
}
```

Maybe it would be better not to use `auto[$]`.